### DISCUSSION TITLE
Using mozjpeg encoder

### DISCUSSION THREAD

**Discussion by TimurSadorov** [SixLabors/ImageSharp#2711] [Q&A]
Hello! I need to convert an image using an mozjpeg encoder. I tried to figure it out, but it didn't work out. How can this be done or is there no such function now?

**Comment by antonfirsov** [comment]
I'm wondering why you prefer mozjpeg over libjpeg? If libjpeg is fine, you can [utilize it with ImageMagick.NET](https://github.com/SixLabors/ImageSharp/blob/main/tests/ImageSharp.Tests/TestUtilities/ReferenceCodecs/MagickReferenceDecoder.cs), (unlike in the sample, you would need to implement an encoder). If mozjpeg is a strict requirement, you need a managed wrapper around it (for example [MozJpegSharp](https://github.com/georg-jung/MozJpegSharp)) and build an `ImageEncoder` with the wrapper.

**Reply by antonfirsov** [reply]
Addition: implementing an `ImageEncoder` is not strictly required for your use case, you can just [iterate through the pixel data](https://docs.sixlabors.com/articles/imagesharp/pixelbuffers.html) and feed it to the encoder logic.

**Reply by TimurSadorov** [reply]
> I'm wondering why you prefer mozjpeg over libjpeg? 

Because mozjpeg compresses the image better:)

**Reply by JimBobSquarePants** [reply]
We don’t use either.
